virtool/samples/data.py

```diff
@@ -158,23 +159,27 @@ async def find(
         )
 
     async def get(self, sample_id: str) -> Sample:
-        document = await self._db.samples.find_one({"_id": sample_id})
+        documents = await self._db.samples.aggregate(
+            [
+                {"$match": {"_id": sample_id}},
+                *lookup_nested_user_by_id(local_field="user.id"),
+                *lookup_nested_subtractions(local_field="subtractions"),
+                *lookup_caches(local_field="_id"),
+            ]
+        ).to_list(length=1)
 
-        if not document:
+        document = None
+
+        if (documents is None) or (len(documents) == 0):
```

Comment:
Were you running into problems where `Documents` was `None`? I Think `to_list` always returns a list even if it is an empty list